env/env.go

```diff
@@ -0,0 +1,58 @@
+/*Package env provides functions to test code that read environment variables
+ */
+package env
+
+import (
+	"os"
+	"strings"
+
+	"github.com/stretchr/testify/require"
+)
+
+// Patch changes the value of an environment variable, and returns a
+// function which will reset the the value of that variable back to the
+// previous state.
+func Patch(t require.TestingT, key, value string) func() {
+	oldValue, ok := os.LookupEnv(key)
+	require.NoError(t, os.Setenv(key, value))
+	return func() {
+		if !ok {
+			require.NoError(t, os.Unsetenv(key))
+			return
+		}
+		require.NoError(t, os.Setenv(key, oldValue))
+	}
+}
+
+// PatchAll sets the environment to env, and returns a function which will
+// reset the environment back to the previous state.
+func PatchAll(t require.TestingT, env map[string]string) func() {
```

Comment:
Should it be `Set` instead (because we "set" the all environment variable)

Reply:
It does do more than set because it also returns a function to restore the state. I think in many languages this is called "patch".

The example usage looks something like this :

```go
	defer PatchAll(t, map[string]string{
		"ONE": "FOO",
		"TWO": "BAR",
	})()
```

I think set might not convey that it will also unset them.

Reply:
Fair enough 👼